src/Admin.php

```diff
@@ -277,4 +298,44 @@ public static function wp_dropdown_users_args(array $query_args) {
     return $query_args;
   }
 
+  /**
+   * @implements wp_revisions_to_keep
+   *
+   * Revisions of other post types are not limited and only cleaned up by a cron
+   * job.
+   *
+   * @see Admin::cron_revision_cleanup()
+   */
+  public static function wp_revisions_to_keep($num, \WP_Post $post) {
+    // Revisions for static pages (such as about or imprint pages) should be
+    // retained infinitely.
+    if ($post->post_type === 'page') {
+      $num = -1;
+    }
+    return $num;
+  }
+
+  /**
+   * Cron event callback to clean up obsolete revisions.
+   *
+   * @param int $limit
+   *   (optional) The maximum amount of revisions to delete. Defaults to 100 in
+   *   order to not slow down and exceed cron execution time.
+   */
+  public static function cron_revision_cleanup($limit = 100) {
+    global $wpdb;
+
+    // Revisions of posts that were last modified a long time ago (3 months) are
+    // no longer necessary and can be cleaned up.
+    $revision_ids = $wpdb->get_col($wpdb->prepare("SELECT revision.ID
+FROM $wpdb->posts revision
+INNER JOIN $wpdb->posts parent ON parent.ID = revision.post_parent AND parent.post_type = %s
+WHERE revision.post_type = 'revision' AND revision.post_modified_gmt < %s AND revision.post_name NOT LIKE '%autosave%'
```

Comment:
I was wondering why you would use the published post date instead of the modified date as it will affect "evergreen" articles as well, as state in your notes.

> - For dynamic content (post type "post"), store only revisions within 6 months after article publishing date. (0)

> Notes
> - This also affects edits of evergreen articles — editors need to make sure to update the publishing date in such cases.

… Just seen you stated this out in one of your comments already.